### ISSUE TITLE
When using older version of xUnit test cases are not shown

### ISSUE THREAD

**Issue by wcc-neal** [axodox/AxoTools#101]
Background:
----
VS 2017 Community  - 15.0.0+26228.10
AxoCover 1.1.229.0 (Issue also present in 1.1.228.0)
Resharper 2017.1.2 (Issue also present when I suspend it)

Steps Taken:
----

1. Open my solution, containing a number of projects, one of which is my test project, containing xUnit tests.
2. Rebuild solution
3. View AxoCover window

Expected Result:
----
AxoCover should list my tests

Actual Result:
----
"Hmm... AxoCover found no unit tests in this solution."
Also, in the output window from AxoCover, I see:

```> Discovering tests...
| Runner version is 1.1.229.0.
| File redirection is enabled for the following files:
| C:\USERS\NMCCONACHIE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\15.0_7AFA8198\EXTENSIONS\jd51ug4j.kes\xUnitAdapter\xunit.abstractions.dll
| C:\USERS\NMCCONACHIE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\15.0_7AFA8198\EXTENSIONS\jd51ug4j.kes\xUnitAdapter\xunit.assert.dll
| C:\USERS\NMCCONACHIE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\15.0_7AFA8198\EXTENSIONS\jd51ug4j.kes\xUnitAdapter\xunit.core.dll
| C:\USERS\NMCCONACHIE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\15.0_7AFA8198\EXTENSIONS\jd51ug4j.kes\xUnitAdapter\xunit.execution.desktop.dll
| C:\USERS\NMCCONACHIE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\15.0_7AFA8198\EXTENSIONS\jd51ug4j.kes\xUnitAdapter\xunit.runner.reporters.net452.dll
| C:\USERS\NMCCONACHIE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\15.0_7AFA8198\EXTENSIONS\jd51ug4j.kes\xUnitAdapter\xunit.runner.utility.net35.dll
| C:\USERS\NMCCONACHIE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\15.0_7AFA8198\EXTENSIONS\jd51ug4j.kes\xUnitAdapter\xunit.runner.visualstudio.testadapter.dll
| Setting up file redirection hooks...
| File redirection hooks are enabled.
>> Loading assembly from C:\USERS\NMCCONACHIE\APPDATA\LOCAL\MICROSOFT\VISUALSTUDIO\15.0_7AFA8198\EXTENSIONS\jd51ug4j.kes\xUnitAdapter\xunit.runner.visualstudio.testadapter.dll...
|| Loaded discoverer: Xunit.Runner.VisualStudio.TestAdapter.VsTestRunner.
<< Assembly loaded.
>> Running discoverer: Xunit.Runner.VisualStudio.TestAdapter.VsTestRunner...
|| Checking C:\dev\projects\tc-members\TradeclubMembers.Tests\bin\Debug\TradeclubMembers.Tests.dll...
   [xUnit.net 00:00:00.4918944]   Discovering: TradeclubMembers.Tests
   [xUnit.net 00:00:00.5544803]   Discovered:  TradeclubMembers.Tests
<< Discoverer finished.
| File redirection rules are cleared.
< Test discovery finished.
Test discovery finished.```

Both the native VS Test Explorer, and (after I re-enable it) the ReSharper Unit Test Sessions window show my tests.

**Comment by axodox** [comment]
Is this project using the full .Net Framework? - .Net Core is not yet supported by AxoCover
If it us the full framework, please a sample project and attach it.

**Comment by wcc-neal** [comment]
Yes, it is the full .Net Framework - I am not in a position to divulge the full source code, but I can try to create a minimal project that has this issue, and attach that.

**Comment by wcc-neal** [comment]
Ahh, in the process of creating the test case, I've updated my xUnit nuget package from 2.1.0 to the latest (2.2.0) - now AxoCover is working again.

Thanks for taking the time to respond - this is resolved for me now :)

**Comment by axodox** [comment]
I will reopen this, it is intended to work with old version as well, thanks for the info.

**Comment by wcc-neal** [comment]
In this case, here's a .zip file of a minimal solution which reproduces the bug. 
[TestCase.zip](https://github.com/axodox/AxoTools/files/1158993/TestCase.zip)

**Comment by axodox** [comment]
Thanks, I already reproduced it though :)

**Comment by axodox** [comment]
It will likely take a couple of weeks though, depending on how much free time I have, and how much of it will this take to fix.

**Comment by wcc-neal** [comment]
No worries - thanks for the quick response :-) In my specific case, upgrading xUnit is not a problem.

**Comment by axodox** [comment]
I looked at this the other day, and it seems there were some significant changes between these xunit versions, I unfortunately have no free time to maintain so many branches. I will try to get my changes to be merged to the official xunit and nunit, so no AxoCover specific ports will be needed. In the meantime I will add some warning to explain the issue to users when encountered.

**Comment by axodox** [comment]
Yes, so apparently 2.1, 2.2 and 2.3 are not compatible with each other. Unfortunately I still need my no shadow copy workaround for OpenCover to work, which is not exposed otherwise by xUnit VS plug-in - which I have reused - and such change suggested by others had been rejected multiple times (for issues I don't agree with). So I need to keep the fork.

I decided to show the exact test framework versions / platform I support like this:
![image](https://user-images.githubusercontent.com/22882208/31585925-3724562a-b1ca-11e7-8558-daf60e0953f1.png)

I will update these to latest time to time when people ask for it, but I really don't have the resources to support many versions (or platforms) by myself for free. So I will stick with supporting what I actually need at work and some of the ones people ask which are still fit the timeframe I am willing to put into it.
